python/proton/_message.py

```diff
@@ -90,13 +90,16 @@ def _check(self, err):
 
     def _check_property_keys(self):
         for k in self.properties.keys():
-            if isinstance(k, unicode):
-                # py2 unicode, py3 str (via hack definition)
+            # Check for string type. (py2: unicode, py3: str via type hack above)
+            # String subclasses symbol and char are excluded
+            # (But so are other string subclasses that would be encoded as type string!)
+            if type(k) == unicode:
```

Comment:
I did it this way on the understanding that we convert ALL unicode subclasses, including symbol and char, and that this would add to an easier user experience. Certainly it would be more "technically correct" to exclude symbol and char though.

Reply:
After discussion, we think that AMQP subclasses of string should be excluded from the conversion. Thus, symbol and char will cause an exception, but other user-defined subclasses will be converted.